test/TestPercentageMath.sol

```diff
@@ -201,6 +204,17 @@ contract TestPercentageMath is Test {
         PercentageMath.weightedAvg(x, y, percentage);
     }
 
+    function testWeightedAvgOutOfBounds() public {
+        uint256 x = 5000;
+        uint256 y = 5000;
+        uint256 percentage = 1;
+
+        uint256 weightedAvg = PercentageMath.weightedAvg(x, y, percentage);
+
+        assertLe(x, weightedAvg);
+        assertLe(weightedAvg, y);
+    }
```

Comment:
As an example, the idea of this test sounds really good on the other hand. Once we fixed the behavior in this PR we want to maintain it forever (or at least it should cost a bit to work around it)

Reply:
yes I agree